exp/controllers/awsmachinepool_controller.go

```diff
@@ -415,6 +448,164 @@ func (r *AWSMachinePoolReconciler) reconcileDelete(machinePoolScope *scope.Machi
 	return nil
 }
 
+func reconcileDeleteAWSMachines(ctx context.Context, mp *expclusterv1.MachinePool, clusterScope cloud.ClusterScoper, client client.Client) error {
+	awsMachineList, err := getAWSMachines(ctx, mp, client)
+	if err != nil {
+		return err
+	}
+	for i := range awsMachineList.Items {
+		awsMachine := awsMachineList.Items[i]
+		if !awsMachine.DeletionTimestamp.IsZero() {
+			// delete the owner Machine resource for the AWSMachine so that CAPI can clean up gracefully
+			machine, err := util.GetOwnerMachine(ctx, client, awsMachine.ObjectMeta)
+			if err != nil {
+				clusterScope.Warn("failed to get owner machine", "awsmachine", awsMachine, "machinepool", mp.Name, "namespace", mp.Namespace)
```

Comment:
So I recalled why I chose the cluster scope. If we use the machinePoolScope here, the `AWSManagedMachinePool` controller can not re-use and call these methods. The `ManagedMachinePool` type is not a compatible interface.

What do you propose to do about that? Do we need to have duplicate code for all these methods in the `AWSManagedMachinePool` controller? I had hoped to avoid that.